### PR TITLE
[X86][StrictFP] Add widening support for STRICT_FMIN/STRICT_FMAX

### LINKED ISSUE #119422: X86 type legalizer crash on X86ISD::STRICT_FMIN node
Getting the following error on a short C testcase with -frounding-math flag:
ReplaceNodeResults: t15: v2f32,ch = X86ISD::STRICT_FMIN t0, t3, t5, example.c:6:14
Do not know how to custom type legalize this operation!
UNREACHABLE executed at /root/llvm-project/llvm/lib/Target/X86/X86ISelLowering.cpp:33268!

reduced .c reproducer
https://godbolt.org/z/5zjarsjv5

reduced .ll reproducer:
https://godbolt.org/z/7G4rcWq4G

After "git bisecting" it seems that this crash started only after introducing commit https://github.com/llvm/llvm-project/commit/9882b35a3a3e46d749b801bd0b98c3d90af6006c

### PR DESCRIPTION
**Pull request by phoebewang**
Fixes: #119422

### CODE REVIEW

File: llvm/lib/Target/X86/X86ISelLowering.cpp [hunk 1]
```diff
@@ -33420,15 +33420,27 @@ void X86TargetLowering::ReplaceNodeResults(SDNode *N,
   case X86ISD::FMINC:
   case X86ISD::FMIN:
   case X86ISD::FMAXC:
-  case X86ISD::FMAX: {
+  case X86ISD::FMAX:
+  case X86ISD::STRICT_FMIN:
+  case X86ISD::STRICT_FMAX: {
     EVT VT = N->getValueType(0);
     assert(VT == MVT::v2f32 && "Unexpected type (!= v2f32) on FMIN/FMAX.");
+    unsigned Opc = N->getOpcode();
```
[the remark this answers is not in the document]
Comment (phoebewang) [reply llvm/lib/Target/X86/X86ISelLowering.cpp:33428]: Done https://github.com/llvm/llvm-project/pull/119494

File: llvm/test/CodeGen/X86/vec-strict-cmp-128.ll [hunk 2]
```diff
@@ -6087,9 +6087,57 @@ define <2 x double> @test_v4f64_ogt2_s(<2 x double> %a, <2 x double> %b) #0 {
...
+; AVX512F-64-LABEL: test_v2f32_ogt2_s:
+; AVX512F-64:       # %bb.0:
+; AVX512F-64-NEXT:    vmaxps %xmm1, %xmm0, %xmm0
+; AVX512F-64-NEXT:    retq
+  %cond = call <2 x i1> @llvm.experimental.constrained.fcmps.v2f32(
+                                               <2 x float> %a, <2 x float> %b, metadata !"ogt",
+                                               metadata !"fpexcept.strict")
+  %res = select <2 x i1> %cond, <2 x float> %a, <2 x float> %b
+  ret <2 x float> %res
+}
```
Comment (RKSimon) [inline llvm/test/CodeGen/X86/vec-strict-cmp-128.ll:6135]: worth adding fmin test as well?

**Review by RKSimon — APPROVED** [review]
LGTM with one (optional) minor- cheers

File: llvm/test/CodeGen/X86/vec-strict-cmp-128.ll (hunk 2 above)
Comment (phoebewang) [reply llvm/test/CodeGen/X86/vec-strict-cmp-128.ll:6135]: Done.

**Pull request merged**